neuralmonkey/decoders/beam_search_decoder.py

```diff
@@ -181,9 +190,22 @@ def cond(*args) -> tf.Tensor:
         dec_loop_state = final_state.decoder_loop_state
         bs_state = final_state.bs_state
 
-        scores = final_state.bs_output.scores.stack()
-        parent_ids = final_state.bs_output.parent_ids.stack()
-        token_ids = final_state.bs_output.token_ids.stack()
+        # Because the "batch" dymension is dynamic and the TensorArrays
+        # are empty during initial call during ensembling we have
+        # to define a default values (the runner itself has to handle them,
+        # if needed
+        scores = tf.cond(
+            tf.equal(final_state.bs_output.scores.size(), 0),
+            lambda: tf.fill([1, self.batch_size, self.beam_size], 0.0),
+            final_state.bs_output.scores.stack)
+        parent_ids = tf.cond(
+            tf.equal(final_state.bs_output.parent_ids.size(), 0),
+            lambda: tf.fill([1, self.batch_size, self.beam_size], 0),
+            final_state.bs_output.parent_ids.stack)
```

Comment:
tady má bejt taky lambda všude ne? má tu bejt 6x napsáno `lambda` a je to tu jen 3x.

Reply:
Jsou testy co testujou oba výsledky toho condu?

Reply:
Nemá, protože `stack` je metoda, psát
```python
lambda: final_state.bs_output.parent_ids.stack()
```
je divný. Spíš bych přidal `NOTE` tak něco, až se změní `tf.cond` na něco rozumnější a budeme to měnit všude v sedě, tak ať víme, kde se to rozbilo.

Reply:
jo rozumim..  je to zvláštní..

Reply:
Za 'lambda: final_state.bs_output.parent_ids.stack()' ti dokonce pylint vynada. NOTE pridam.